service/src/main/java/com/commercetools/pspadapter/payone/ScheduledJob.java

```diff
@@ -42,17 +41,24 @@ public void execute(final JobExecutionContext context) throws JobExecutionExcept
 
             final Consumer<Payment> paymentConsumer = payment -> {
                 try {
-                    final PaymentWithCartLike paymentWithCartLike = queryExecutor.getPaymentWithCartLike(payment.getId(), CompletableFuture.completedFuture(payment));
+                    final PaymentWithCartLike paymentWithCartLike = queryExecutor
+                        .getPaymentWithCartLike(payment.getId(), CompletableFuture.completedFuture(payment));
                     paymentDispatcher.dispatchPayment(paymentWithCartLike);
                 } catch (final NoCartLikeFoundException ex) {
-                    LOG.debug(LOG_PREFIX + " Could not dispatch payment with ID \"{}\": {}",
-                            tenantFactory.getTenantName(),  payment.getId(), ex.getMessage());
+                    LOG.error(
+                        createTenantKeyValue(tenantFactory.getTenantName()),
+                        format("Could not dispatch payment with id '%s'", payment.getId()),
+                        ex);
                 } catch (final ConcurrentModificationException ex) {
-                    LOG.info(LOG_PREFIX + " Could not dispatch payment with ID \"{}\": The payment is currently processed by someone else.",
-                            tenantFactory.getTenantName(), payment.getId());
-                } catch (final Throwable ex) {
-                    LOG.error(LOG_PREFIX + " Error dispatching payment with ID \"{}\"",
-                            tenantFactory.getTenantName(), payment.getId(), ex);
+                    LOG.info(
+                        createTenantKeyValue(tenantFactory.getTenantName()),
+                        format("Could not dispatch payment with id '%s': The payment is currently processed by someone else.", payment.getId()),
```

Comment:
We don't need this. Why do we have it @butenkor?

Reply:
I guess, it can be removed. Are you going just to swallow here?

Reply:
No, it will be caught with the more generic Exception in the catch block below and it will be logged as an error with its stack trace: cbc2a81

Would this be a problem?